File: CPP/saikr/NoC.cc
/**
 * Created by Xiaozhong on 2021/10/31.
 * Copyright (c) 2021/10/31 Xiaozhong. All rights reserved.
 */
#include "iostream"
#include "vector"
#include "algorithm"

using namespace std;

int main() {
    int n = 0;
    cin >> n;
    vector<int> record(n);
    for (int i = 0; i < n; ++i) {
        cin >> record[i];
    }
    sort(record.begin(), record.end());
    cout << record[1];
}

File: Java/src/saikr/NoC.java
/*
 * Copyright (c) 2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

package saikr;

import java.util.Arrays;
import java.util.Scanner;

public class NoC {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int curr = Integer.MAX_VALUE, prev = Integer.MAX_VALUE;
        for (int i = 0; i < n; ++i) {
            int num = scanner.nextInt();
            if (curr == Integer.MAX_VALUE) curr = num;
            else if (prev == Integer.MAX_VALUE) {
                prev = Math.max(num, curr);
                curr = Math.min(num, curr);
            } else {
                if (num < curr) {
                    prev = curr;
                    curr = num;
                } else if (num < prev) prev = num;
            }
        }
        System.out.println(prev);
    }
}
